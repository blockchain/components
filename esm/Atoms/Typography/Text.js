function _objectWithoutProperties(source, excluded) { if (source == null) return {}; var target = _objectWithoutPropertiesLoose(source, excluded); var key, i; if (Object.getOwnPropertySymbols) { var sourceSymbolKeys = Object.getOwnPropertySymbols(source); for (i = 0; i < sourceSymbolKeys.length; i++) { key = sourceSymbolKeys[i]; if (excluded.indexOf(key) >= 0) continue; if (!Object.prototype.propertyIsEnumerable.call(source, key)) continue; target[key] = source[key]; } } return target; }

function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }

function _templateObject() {
  var data = _taggedTemplateLiteral(["\n  font-family: ", ";\n  font-weight: ", ";\n  font-size: ", ";\n  text-transform: ", ";\n  font-style: ", ";\n  color: ", ";\n  opacity: ", ";\n"]);

  _templateObject = function _templateObject() {
    return data;
  };

  return data;
}

function _taggedTemplateLiteral(strings, raw) { if (!raw) { raw = strings.slice(0); } return Object.freeze(Object.defineProperties(strings, { raw: { value: Object.freeze(raw) } })); }

import React from 'react';
import styled from 'styled-components';
import PropTypes from 'prop-types';
var Wrapper = styled.span(_templateObject(), function (props) {
  return props.theme['fontPrimary'];
}, function (props) {
  return props.weight;
}, function (props) {
  return props.size;
}, function (props) {
  return props.uppercase ? 'uppercase' : props.capitalize ? 'capitalize' : 'none';
}, function (props) {
  return props.italic ? 'italic' : 'normal';
}, function (props) {
  return props.theme['textColor'];
}, function (props) {
  return props.opacity;
});

var Text = function Text(_ref) {
  var children = _ref.children,
      rest = _objectWithoutProperties(_ref, ["children"]);

  return React.createElement(Wrapper, rest, children);
};

export var propTypes = {
  capitalize: PropTypes.bool,
  children: PropTypes.node,
  italic: PropTypes.bool,
  opacity: PropTypes.number,
  size: PropTypes.string,
  uppercase: PropTypes.bool,
  weight: PropTypes.oneOf([100, 200, 300, 400, 500, 600, 700])
};
export var defaultProps = {
  capitalize: false,
  italic: false,
  opacity: 1,
  size: '14px',
  uppercase: false,
  weight: 400
};
Text.propTypes = propTypes;
Text.defaultProps = defaultProps;
export default Text;