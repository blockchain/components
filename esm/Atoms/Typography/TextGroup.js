function _objectWithoutProperties(source, excluded) { if (source == null) return {}; var target = _objectWithoutPropertiesLoose(source, excluded); var key, i; if (Object.getOwnPropertySymbols) { var sourceSymbolKeys = Object.getOwnPropertySymbols(source); for (i = 0; i < sourceSymbolKeys.length; i++) { key = sourceSymbolKeys[i]; if (excluded.indexOf(key) >= 0) continue; if (!Object.prototype.propertyIsEnumerable.call(source, key)) continue; target[key] = source[key]; } } return target; }

function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }

function _templateObject() {
  var data = _taggedTemplateLiteral(["\n  text-align: ", ";\n\n  & > * {\n    display: ", ";\n    margin-right: ", ";\n  }\n"]);

  _templateObject = function _templateObject() {
    return data;
  };

  return data;
}

function _taggedTemplateLiteral(strings, raw) { if (!raw) { raw = strings.slice(0); } return Object.freeze(Object.defineProperties(strings, { raw: { value: Object.freeze(raw) } })); }

import React from 'react';
import styled from 'styled-components';
import PropTypes from 'prop-types';
var BaseTextGroup = styled.div(_templateObject(), function (props) {
  return props.align;
}, function (props) {
  return props.inline ? 'inline' : 'block';
}, function (props) {
  return props.inline ? '5px' : '0';
});

var TextGroup = function TextGroup(_ref) {
  var children = _ref.children,
      rest = _objectWithoutProperties(_ref, ["children"]);

  return React.createElement(BaseTextGroup, rest, children);
};

TextGroup.propTypes = {
  align: PropTypes.oneOf(['justify', 'left', 'right', 'center']),
  children: PropTypes.node,
  inline: PropTypes.bool
};
TextGroup.defaultProps = {
  align: 'justify',
  inline: true
};
export default TextGroup;