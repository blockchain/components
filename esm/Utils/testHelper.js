import React from 'react';
import { shallow, mount } from 'enzyme';
import { ThemeProvider } from 'styled-components';
import theme from '../Tools/theme';

function wrapWithTheme(fn, children, options) {
  var wrapper = fn(React.createElement(ThemeProvider, {
    theme: theme
  }, children), options);
  return wrapper[fn.name]({
    context: wrapper.instance().getChildContext()
  });
}

export function shallowWithTheme() {
  return wrapWithTheme.apply(void 0, [shallow].concat(Array.prototype.slice.call(arguments)));
}
export function mountWithTheme() {
  return wrapWithTheme.apply(void 0, [mount].concat(Array.prototype.slice.call(arguments)));
}