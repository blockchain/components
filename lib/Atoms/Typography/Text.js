"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = exports.defaultProps = exports.propTypes = void 0;

var _react = _interopRequireDefault(require("react"));

var _styledComponents = _interopRequireDefault(require("styled-components"));

var _propTypes = _interopRequireDefault(require("prop-types"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _objectWithoutProperties(source, excluded) { if (source == null) return {}; var target = _objectWithoutPropertiesLoose(source, excluded); var key, i; if (Object.getOwnPropertySymbols) { var sourceSymbolKeys = Object.getOwnPropertySymbols(source); for (i = 0; i < sourceSymbolKeys.length; i++) { key = sourceSymbolKeys[i]; if (excluded.indexOf(key) >= 0) continue; if (!Object.prototype.propertyIsEnumerable.call(source, key)) continue; target[key] = source[key]; } } return target; }

function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }

function _templateObject() {
  var data = _taggedTemplateLiteral(["\n  font-family: ", ";\n  font-weight: ", ";\n  font-size: ", ";\n  text-transform: ", ";\n  font-style: ", ";\n  color: ", ";\n  opacity: ", ";\n"]);

  _templateObject = function _templateObject() {
    return data;
  };

  return data;
}

function _taggedTemplateLiteral(strings, raw) { if (!raw) { raw = strings.slice(0); } return Object.freeze(Object.defineProperties(strings, { raw: { value: Object.freeze(raw) } })); }

var Wrapper = _styledComponents.default.span(_templateObject(), function (props) {
  return props.theme['fontPrimary'];
}, function (props) {
  return props.weight;
}, function (props) {
  return props.size;
}, function (props) {
  return props.uppercase ? 'uppercase' : props.capitalize ? 'capitalize' : 'none';
}, function (props) {
  return props.italic ? 'italic' : 'normal';
}, function (props) {
  return props.theme['textColor'];
}, function (props) {
  return props.opacity;
});

var Text = function Text(_ref) {
  var children = _ref.children,
      rest = _objectWithoutProperties(_ref, ["children"]);

  return _react.default.createElement(Wrapper, rest, children);
};

var propTypes = {
  capitalize: _propTypes.default.bool,
  children: _propTypes.default.node,
  italic: _propTypes.default.bool,
  opacity: _propTypes.default.number,
  size: _propTypes.default.string,
  uppercase: _propTypes.default.bool,
  weight: _propTypes.default.oneOf([100, 200, 300, 400, 500, 600, 700])
};
exports.propTypes = propTypes;
var defaultProps = {
  capitalize: false,
  italic: false,
  opacity: 1,
  size: '14px',
  uppercase: false,
  weight: 400
};
exports.defaultProps = defaultProps;
Text.propTypes = propTypes;
Text.defaultProps = defaultProps;
var _default = Text;
exports.default = _default;