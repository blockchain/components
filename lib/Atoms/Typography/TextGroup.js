"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _styledComponents = _interopRequireDefault(require("styled-components"));

var _propTypes = _interopRequireDefault(require("prop-types"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _objectWithoutProperties(source, excluded) { if (source == null) return {}; var target = _objectWithoutPropertiesLoose(source, excluded); var key, i; if (Object.getOwnPropertySymbols) { var sourceSymbolKeys = Object.getOwnPropertySymbols(source); for (i = 0; i < sourceSymbolKeys.length; i++) { key = sourceSymbolKeys[i]; if (excluded.indexOf(key) >= 0) continue; if (!Object.prototype.propertyIsEnumerable.call(source, key)) continue; target[key] = source[key]; } } return target; }

function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }

function _templateObject() {
  var data = _taggedTemplateLiteral(["\n  text-align: ", ";\n\n  & > * {\n    display: ", ";\n    margin-right: ", ";\n  }\n"]);

  _templateObject = function _templateObject() {
    return data;
  };

  return data;
}

function _taggedTemplateLiteral(strings, raw) { if (!raw) { raw = strings.slice(0); } return Object.freeze(Object.defineProperties(strings, { raw: { value: Object.freeze(raw) } })); }

var BaseTextGroup = _styledComponents.default.div(_templateObject(), function (props) {
  return props.align;
}, function (props) {
  return props.inline ? 'inline' : 'block';
}, function (props) {
  return props.inline ? '5px' : '0';
});

var TextGroup = function TextGroup(_ref) {
  var children = _ref.children,
      rest = _objectWithoutProperties(_ref, ["children"]);

  return _react.default.createElement(BaseTextGroup, rest, children);
};

TextGroup.propTypes = {
  align: _propTypes.default.oneOf(['justify', 'left', 'right', 'center']),
  children: _propTypes.default.node,
  inline: _propTypes.default.bool
};
TextGroup.defaultProps = {
  align: 'justify',
  inline: true
};
var _default = TextGroup;
exports.default = _default;