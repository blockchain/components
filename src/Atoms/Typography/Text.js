import React from 'react'
import styled from 'styled-components'
import PropTypes from 'prop-types'
import { keysIn } from 'ramda'
import theme from '../../Tools/theme'

const BaseText = styled.span`
  font-family: 'Montserrat', helvetica, sans serif;
  font-weight: ${(props) => props.weight};
  font-size: ${(props) => props.size};
  text-transform: ${(props) =>
    props.uppercase ? 'uppercase' : props.capitalize ? 'capitalize' : 'none'};
  font-style: ${(props) => (props.italic ? 'italic' : 'normal')};
  color: ${(props) => props.theme[props.color]};
  opacity: ${(props) => props.opacity};
`

const Text = ({
  children,
  weight,
  size,
  uppercase,
  capitalize,
  italic,
  color,
  opacity,
  ...rest
}) => (
  <BaseText
    capitalize={capitalize}
    color={color}
    italic={italic}
    opacity={opacity}
    size={size}
    uppercase={uppercase}
    weight={weight}
    {...rest}
  >
    {children}
  </BaseText>
)

Text.propTypes = {
  capitalize: PropTypes.bool,
  children: PropTypes.node,
  color: PropTypes.oneOf(keysIn(theme)),
  italic: PropTypes.bool,
  opacity: PropTypes.number,
  size: PropTypes.string,
  uppercase: PropTypes.bool,
  weight: PropTypes.oneOf([100, 200, 300, 400, 500, 600, 700, 800, 900]),
}

Text.defaultProps = {
  capitalize: false,
  color: 'gray-5',
  italic: false,
  opacity: 1,
  size: '14px',
  uppercase: false,
  weight: 300,
}

export default Text
