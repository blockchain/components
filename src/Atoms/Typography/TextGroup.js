// @flow strict
import PropTypes from 'prop-types'
import * as React from 'react'
import styled from 'styled-components'

import { prop } from '../../Tools/interpolation'

export type PropsType = {
  +align?: 'center' | 'justify' | 'left' | 'right',
  +capitalize?: boolean,
  +children?: React.Node,
  +inline?: boolean,
}

const Wrapper = styled.div`
  text-align: ${prop<PropsType>('align')};

  & > * {
    display: ${(props: PropsType) =>
      props.inline === true ? 'inline' : 'block'};
    margin-right: ${(props: PropsType) =>
      props.inline === true ? '0.4rem' : '0'};
  }
`

const TextGroup = ({ children, ...rest }: PropsType) => (
  <Wrapper {...rest}>{children}</Wrapper>
)

TextGroup.propTypes = {
  align: PropTypes.oneOf(['justify', 'left', 'right', 'center']),
  capitalize: PropTypes.bool,
  children: PropTypes.node,
  inline: PropTypes.bool,
}

TextGroup.defaultProps = {
  align: 'justify',
  inline: true,
}

export default TextGroup
