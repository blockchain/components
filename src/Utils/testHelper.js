import React from 'react'
import { shallow, mount } from 'enzyme'
import { ThemeProvider } from 'styled-components'
import theme from '../Tools/theme'

function wrapWithTheme(fn, children, options) {
  const wrapper = fn(
    <ThemeProvider theme={theme}>{children}</ThemeProvider>,
    options,
  )

  return wrapper[fn.name]({
    context: wrapper.instance().getChildContext(),
  })
}

export function shallowWithTheme() {
  return wrapWithTheme(shallow, ...arguments)
}

export function mountWithTheme() {
  return wrapWithTheme(mount, ...arguments)
}
