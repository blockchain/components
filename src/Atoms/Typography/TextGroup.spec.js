import React from 'react'
import { mountWithTheme } from '../../Utils/testHelper'

import Text from './Text'
import TextGroup from './TextGroup'

const Texts = () => (
  <>
    <Text>
      This is my first text. This is my first text. This is my first text. This
      is my first text.
    </Text>
    <Text>
      This is my second text. This is my second text. This is my second text.
      This is my second text.
    </Text>
    <Text>
      This is my third text. This is my third text. This is my third text. This
      is my third text.
    </Text>
    <Text>
      This is my fourth text. This is my fourth text. This is my fourth text.
      This is my fourth text.
    </Text>
    <Text>
      This is my fifth text. This is my fifth text. This is my fifth text. This
      is my fifth text.
    </Text>
    <Text>
      This is my sixth text. This is my sixth text. This is my sixth text. This
      is my sixth text.
    </Text>
    <Text>
      This is my seventh text. This is my seventh text. This is my seventh text.
      This is my seventh text.
    </Text>
  </>
)

describe('TextGroup', () => {
  it('mount render correctly', () => {
    const tree = mountWithTheme(
      <TextGroup>
        <Texts />
      </TextGroup>,
    )

    expect(tree.find(TextGroup)).toMatchSnapshot()
  })

  it('mount render correctly with align', () => {
    const tree = mountWithTheme(
      <TextGroup align="right">
        <Texts />
      </TextGroup>,
    )

    expect(tree.find(TextGroup)).toMatchSnapshot()
  })

  it('mount render correctly with inline', () => {
    const tree = mountWithTheme(
      <TextGroup inline={false}>
        <Texts />
      </TextGroup>,
    )

    expect(tree.find(TextGroup)).toMatchSnapshot()
  })
})
