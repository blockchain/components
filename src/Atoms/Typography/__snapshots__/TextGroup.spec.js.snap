// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextGroup mount render correctly 1`] = `
.c1 {
  font-family: 'IBM Plex Sans','Helvetica Neue',Arial,sans-serif;
  font-weight: 400;
  font-size: 1rem;
  text-transform: none;
  font-style: normal;
  color: #545456;
  opacity: 1;
}

.c0 {
  text-align: justify;
}

.c0 > * {
  display: inline;
  margin-right: 0.4rem;
}

<TextGroup
  align="justify"
  inline={true}
>
  <styled.div
    align="justify"
    inline={true}
  >
    <div
      className="c0"
    >
      <Texts>
        <Text
          capitalize={false}
          italic={false}
          opacity={1}
          size="1rem"
          uppercase={false}
          weight={400}
        >
          <styled.span
            capitalize={false}
            italic={false}
            opacity={1}
            size="1rem"
            uppercase={false}
            weight={400}
          >
            <span
              className="c1"
              opacity={1}
              size="1rem"
            >
              This is my first text. This is my first text. This is my first text. This is my first text.
            </span>
          </styled.span>
        </Text>
        <Text
          capitalize={false}
          italic={false}
          opacity={1}
          size="1rem"
          uppercase={false}
          weight={400}
        >
          <styled.span
            capitalize={false}
            italic={false}
            opacity={1}
            size="1rem"
            uppercase={false}
            weight={400}
          >
            <span
              className="c1"
              opacity={1}
              size="1rem"
            >
              This is my second text. This is my second text. This is my second text. This is my second text.
            </span>
          </styled.span>
        </Text>
        <Text
          capitalize={false}
          italic={false}
          opacity={1}
          size="1rem"
          uppercase={false}
          weight={400}
        >
          <styled.span
            capitalize={false}
            italic={false}
            opacity={1}
            size="1rem"
            uppercase={false}
            weight={400}
          >
            <span
              className="c1"
              opacity={1}
              size="1rem"
            >
              This is my third text. This is my third text. This is my third text. This is my third text.
            </span>
          </styled.span>
        </Text>
        <Text
          capitalize={false}
          italic={false}
          opacity={1}
          size="1rem"
          uppercase={false}
          weight={400}
        >
          <styled.span
            capitalize={false}
            italic={false}
            opacity={1}
            size="1rem"
            uppercase={false}
            weight={400}
          >
            <span
              className="c1"
              opacity={1}
              size="1rem"
            >
              This is my fourth text. This is my fourth text. This is my fourth text. This is my fourth text.
            </span>
          </styled.span>
        </Text>
        <Text
          capitalize={false}
          italic={false}
          opacity={1}
          size="1rem"
          uppercase={false}
          weight={400}
        >
          <styled.span
            capitalize={false}
            italic={false}
            opacity={1}
            size="1rem"
            uppercase={false}
            weight={400}
          >
            <span
              className="c1"
              opacity={1}
              size="1rem"
            >
              This is my fifth text. This is my fifth text. This is my fifth text. This is my fifth text.
            </span>
          </styled.span>
        </Text>
        <Text
          capitalize={false}
          italic={false}
          opacity={1}
          size="1rem"
          uppercase={false}
          weight={400}
        >
          <styled.span
            capitalize={false}
            italic={false}
            opacity={1}
            size="1rem"
            uppercase={false}
            weight={400}
          >
            <span
              className="c1"
              opacity={1}
              size="1rem"
            >
              This is my sixth text. This is my sixth text. This is my sixth text. This is my sixth text.
            </span>
          </styled.span>
        </Text>
        <Text
          capitalize={false}
          italic={false}
          opacity={1}
          size="1rem"
          uppercase={false}
          weight={400}
        >
          <styled.span
            capitalize={false}
            italic={false}
            opacity={1}
            size="1rem"
            uppercase={false}
            weight={400}
          >
            <span
              className="c1"
              opacity={1}
              size="1rem"
            >
              This is my seventh text. This is my seventh text. This is my seventh text. This is my seventh text.
            </span>
          </styled.span>
        </Text>
      </Texts>
    </div>
  </styled.div>
</TextGroup>
`;

exports[`TextGroup mount render correctly with align 1`] = `
.c1 {
  font-family: 'IBM Plex Sans','Helvetica Neue',Arial,sans-serif;
  font-weight: 400;
  font-size: 1rem;
  text-transform: none;
  font-style: normal;
  color: #545456;
  opacity: 1;
}

.c0 {
  text-align: right;
}

.c0 > * {
  display: inline;
  margin-right: 0.4rem;
}

<TextGroup
  align="right"
  inline={true}
>
  <styled.div
    align="right"
    inline={true}
  >
    <div
      className="c0"
    >
      <Texts>
        <Text
          capitalize={false}
          italic={false}
          opacity={1}
          size="1rem"
          uppercase={false}
          weight={400}
        >
          <styled.span
            capitalize={false}
            italic={false}
            opacity={1}
            size="1rem"
            uppercase={false}
            weight={400}
          >
            <span
              className="c1"
              opacity={1}
              size="1rem"
            >
              This is my first text. This is my first text. This is my first text. This is my first text.
            </span>
          </styled.span>
        </Text>
        <Text
          capitalize={false}
          italic={false}
          opacity={1}
          size="1rem"
          uppercase={false}
          weight={400}
        >
          <styled.span
            capitalize={false}
            italic={false}
            opacity={1}
            size="1rem"
            uppercase={false}
            weight={400}
          >
            <span
              className="c1"
              opacity={1}
              size="1rem"
            >
              This is my second text. This is my second text. This is my second text. This is my second text.
            </span>
          </styled.span>
        </Text>
        <Text
          capitalize={false}
          italic={false}
          opacity={1}
          size="1rem"
          uppercase={false}
          weight={400}
        >
          <styled.span
            capitalize={false}
            italic={false}
            opacity={1}
            size="1rem"
            uppercase={false}
            weight={400}
          >
            <span
              className="c1"
              opacity={1}
              size="1rem"
            >
              This is my third text. This is my third text. This is my third text. This is my third text.
            </span>
          </styled.span>
        </Text>
        <Text
          capitalize={false}
          italic={false}
          opacity={1}
          size="1rem"
          uppercase={false}
          weight={400}
        >
          <styled.span
            capitalize={false}
            italic={false}
            opacity={1}
            size="1rem"
            uppercase={false}
            weight={400}
          >
            <span
              className="c1"
              opacity={1}
              size="1rem"
            >
              This is my fourth text. This is my fourth text. This is my fourth text. This is my fourth text.
            </span>
          </styled.span>
        </Text>
        <Text
          capitalize={false}
          italic={false}
          opacity={1}
          size="1rem"
          uppercase={false}
          weight={400}
        >
          <styled.span
            capitalize={false}
            italic={false}
            opacity={1}
            size="1rem"
            uppercase={false}
            weight={400}
          >
            <span
              className="c1"
              opacity={1}
              size="1rem"
            >
              This is my fifth text. This is my fifth text. This is my fifth text. This is my fifth text.
            </span>
          </styled.span>
        </Text>
        <Text
          capitalize={false}
          italic={false}
          opacity={1}
          size="1rem"
          uppercase={false}
          weight={400}
        >
          <styled.span
            capitalize={false}
            italic={false}
            opacity={1}
            size="1rem"
            uppercase={false}
            weight={400}
          >
            <span
              className="c1"
              opacity={1}
              size="1rem"
            >
              This is my sixth text. This is my sixth text. This is my sixth text. This is my sixth text.
            </span>
          </styled.span>
        </Text>
        <Text
          capitalize={false}
          italic={false}
          opacity={1}
          size="1rem"
          uppercase={false}
          weight={400}
        >
          <styled.span
            capitalize={false}
            italic={false}
            opacity={1}
            size="1rem"
            uppercase={false}
            weight={400}
          >
            <span
              className="c1"
              opacity={1}
              size="1rem"
            >
              This is my seventh text. This is my seventh text. This is my seventh text. This is my seventh text.
            </span>
          </styled.span>
        </Text>
      </Texts>
    </div>
  </styled.div>
</TextGroup>
`;

exports[`TextGroup mount render correctly with inline 1`] = `
.c1 {
  font-family: 'IBM Plex Sans','Helvetica Neue',Arial,sans-serif;
  font-weight: 400;
  font-size: 1rem;
  text-transform: none;
  font-style: normal;
  color: #545456;
  opacity: 1;
}

.c0 {
  text-align: justify;
}

.c0 > * {
  display: block;
  margin-right: 0;
}

<TextGroup
  align="justify"
  inline={false}
>
  <styled.div
    align="justify"
    inline={false}
  >
    <div
      className="c0"
    >
      <Texts>
        <Text
          capitalize={false}
          italic={false}
          opacity={1}
          size="1rem"
          uppercase={false}
          weight={400}
        >
          <styled.span
            capitalize={false}
            italic={false}
            opacity={1}
            size="1rem"
            uppercase={false}
            weight={400}
          >
            <span
              className="c1"
              opacity={1}
              size="1rem"
            >
              This is my first text. This is my first text. This is my first text. This is my first text.
            </span>
          </styled.span>
        </Text>
        <Text
          capitalize={false}
          italic={false}
          opacity={1}
          size="1rem"
          uppercase={false}
          weight={400}
        >
          <styled.span
            capitalize={false}
            italic={false}
            opacity={1}
            size="1rem"
            uppercase={false}
            weight={400}
          >
            <span
              className="c1"
              opacity={1}
              size="1rem"
            >
              This is my second text. This is my second text. This is my second text. This is my second text.
            </span>
          </styled.span>
        </Text>
        <Text
          capitalize={false}
          italic={false}
          opacity={1}
          size="1rem"
          uppercase={false}
          weight={400}
        >
          <styled.span
            capitalize={false}
            italic={false}
            opacity={1}
            size="1rem"
            uppercase={false}
            weight={400}
          >
            <span
              className="c1"
              opacity={1}
              size="1rem"
            >
              This is my third text. This is my third text. This is my third text. This is my third text.
            </span>
          </styled.span>
        </Text>
        <Text
          capitalize={false}
          italic={false}
          opacity={1}
          size="1rem"
          uppercase={false}
          weight={400}
        >
          <styled.span
            capitalize={false}
            italic={false}
            opacity={1}
            size="1rem"
            uppercase={false}
            weight={400}
          >
            <span
              className="c1"
              opacity={1}
              size="1rem"
            >
              This is my fourth text. This is my fourth text. This is my fourth text. This is my fourth text.
            </span>
          </styled.span>
        </Text>
        <Text
          capitalize={false}
          italic={false}
          opacity={1}
          size="1rem"
          uppercase={false}
          weight={400}
        >
          <styled.span
            capitalize={false}
            italic={false}
            opacity={1}
            size="1rem"
            uppercase={false}
            weight={400}
          >
            <span
              className="c1"
              opacity={1}
              size="1rem"
            >
              This is my fifth text. This is my fifth text. This is my fifth text. This is my fifth text.
            </span>
          </styled.span>
        </Text>
        <Text
          capitalize={false}
          italic={false}
          opacity={1}
          size="1rem"
          uppercase={false}
          weight={400}
        >
          <styled.span
            capitalize={false}
            italic={false}
            opacity={1}
            size="1rem"
            uppercase={false}
            weight={400}
          >
            <span
              className="c1"
              opacity={1}
              size="1rem"
            >
              This is my sixth text. This is my sixth text. This is my sixth text. This is my sixth text.
            </span>
          </styled.span>
        </Text>
        <Text
          capitalize={false}
          italic={false}
          opacity={1}
          size="1rem"
          uppercase={false}
          weight={400}
        >
          <styled.span
            capitalize={false}
            italic={false}
            opacity={1}
            size="1rem"
            uppercase={false}
            weight={400}
          >
            <span
              className="c1"
              opacity={1}
              size="1rem"
            >
              This is my seventh text. This is my seventh text. This is my seventh text. This is my seventh text.
            </span>
          </styled.span>
        </Text>
      </Texts>
    </div>
  </styled.div>
</TextGroup>
`;
